src/libraries/System.Private.CoreLib/src/System/SpanHelpers.cs

```diff
@@ -69,23 +70,26 @@ public static void ClearWithoutReferences(ref byte b, nuint byteLength)
                 nuint stopLoopAtOffset = byteLength & ~(nuint)7;
                 do
                 {
-                    // JIT is expected to coalesce these stores into a single 8-byte store on 64-bit platforms
-                    Unsafe.AddByteOffset(ref Unsafe.As<byte, uint>(ref b), (nint)i) = 0;
-                    Unsafe.AddByteOffset(ref Unsafe.As<byte, uint>(ref b), (nint)i + 4) = 0;
+#if TARGET_64BIT
```

Comment:
Is this really necessary for x86?

https://sharplab.io/#v2:C4LghgzgtgPgAgJgIwFgBQcAMACOSB0ASgK4B2wAllAKb4DCA9lAA4UA21ATgMpcBuFAMbUIAbnRwAzLgTY62AN7psK3NLgAWbAFkkACk7UAZtgBCAT2DVsAIwA02UsQrlsFAJSLlqn9gCqpBBgRrQA6pwUVgFgbBQA5qTUACYAPMRsDKRxAHwGxv6BwbQAgklJFlYA8kZGENTAeSb2bu4OmO7iaD4Avt4qUrha2giNZpbWzU4uwC1eXb6qAUEh+OGR1NGxCclp07mGJktF+KXl49W19aPNHm0dfQtHK2tRpDHxianO5Pv5TyVlCrUC51BoHWwOCjYADU2A0rWw7U6PXQ3SAA===

Reply:
yeah should be fine to leave 8 byte store I guess